app/controllers/rentals_controller.rb

```diff
@@ -102,23 +135,26 @@ def update
 
   # POST /rentals
   def create
-    @rental = Rental.new(rental_params.merge(creator: @current_user))
-
-    @start_date = params['start_date'] || Time.zone.today
-    if @rental.save
-      if params[:amount] && @current_user.has_permission?('rentals', 'cost_adjustment')
-        # find existing financial_transaction and change it
-        @financial_transaction = FinancialTransaction.find_by rental: @rental, transactable_type: Rental.name, transactable_id: @rental.id
-        @financial_transaction.amount = params[:amount]
-        @financial_transaction.save
-      end # if they dont have permission ignore it and we will use default pricing
+    rental = Rental.new rental_params.merge(creator: @current_user)
+
+    # if the specified a custom amount we will create the financial transactions later
+    rental.skip_financial_transactions = params[:amount] && rental.cost != params[:amount] &&
+                                          @current_user.has_permission?('rentals', 'cost_adjustment')
+
+    if rental.save
+      if rental.skip_financial_transactions # use that special pricing, we already verified their perms
+        # create a single custom financial transaction for entire rental
+        FinancialTransaction.create rental: rental, transactable_type: Rental.name, transactable_id: rental.id,
+                                    amount: params[:amount], note_field: "custom rental pricing by #{@current_user.full_name} (#{@current_user.id})"
+      end # else use default pricing
+
       flash[:success] = 'Rental Successfully Reserved'
-      redirect_to(@rental)
-    else # error has problem, cannot rental a error message here
-      set_users_to_assign
-      flash[:warning] = (@rental.item_type.try(:name) || 'Item type') + ' Is Not Available For Specified Dates'
-      @rental.errors.full_messages.each { |e| flash_message :warning, e, :now }
-      render :new
+      redirect_to(rental)
+    else
+      # flash[:warning] = 'Item type is not available for specified dates' # TODO find out the actual error
```

Comment:
Why don't you know already what the actual error is?

Reply:
Right, you have validations in the model.

Reply:
I added the line below which prints out all the errors,  I was questioning why this flash warning had such a specific error message in the past.  I could probably just delete it but I have the suspicion it does not recognize failed reservations.

Reply:
> I added the line below which prints out all the errors

So then this isn't a TODO? It's done?

Reply:
I left that line there because of my suspicion it does not recognize failed reservations because of items not available.